README.md

```diff
@@ -738,90 +738,75 @@ Hazelcast serializes all your objects before sending them to the server. The `un
-
-A sample `DataSerializableFactory` could be implemented as follows:
 
-```C++
-class SampleDataSerializableFactory : public serialization::DataSerializableFactory {
-public:
-    static const int FACTORY_ID = 1000;
-
-    virtual std::auto_ptr<serialization::IdentifiedDataSerializable> create(int32_t classId) {
-        switch (classId) {
-            case 100:
-                return std::auto_ptr<serialization::IdentifiedDataSerializable>(new Employee());
-            default:
-                return std::auto_ptr<serialization::IdentifiedDataSerializable>();
-        }
+namespace hazelcast {
+    namespace client {
+        namespace serialization {
+            template<>
+            struct hz_serializer<Person> : identified_data_serializer {
+                static int32_t get_factory_id() noexcept {
+                    return 1;
+                }
 
-    }
-};
-```
+                static int32_t get_class_id() noexcept {
+                    return 3;
+                }
 
-The last step is to register the `IdentifiedDataSerializableFactory` to the `serialization_config`.
+                static void write_data(const Person &object, hazelcast::client::serialization::object_data_output &out) {
+                    out.write(object.name);
+                    out.write(object.male);
+                    out.write(object.age);
+                }
 
-```C++
-    client_config clientConfig;
-    clientConfig.get_serialization_config().addDataSerializableFactory(SampleDataSerializableFactory::FACTORY_ID,
-                                                                     boost::shared_ptr<serialization::DataSerializableFactory>(
-                                                                             new SampleDataSerializableFactory()));
+                static Person read_data(hazelcast::client::serialization::object_data_input &in) {
+                    return Person{in.read<std::string>(), in.read<bool>(), in.read<int32_t>()};
+                }
+            };
+        }
+    }
+}
 ```
 
-Note that the ID that is passed to the `serialization_config` is same as the `factoryId` that the `Employee` object returns.
+`hz_serializer<Person>` specialization of `hz_serializer` should implement the above four methods, namely `get_factory_id`, `get_class_id`, `write_data`, `read_data`. In case that the object fields are non-public, you can always define struct `hz_serializer<Person>` as friend to your object class. You use the `object_data_output` class methods when serializing the object into binary bytes and you use the `object_data_input` class methods when de-serializing the bytes into the concrete object instance. The factory and class id returned from the associated methods should match the id's defined at the server side (Yes, you need to have the corresponding Java classes at the server side).
```

Comment:
... as a friends to your...

... into binary bytes and the `object_data_input` ...

... and class ID returned from the ...

... should match the IDs defined at ...

... (you need to have the ...